src/main/java/com/lineate/mdzinarishvili/reminderapp/services/ReminderService.java

```diff
@@ -1,4 +1,47 @@
 package com.lineate.mdzinarishvili.reminderapp.services;
 
+import com.lineate.mdzinarishvili.reminderapp.dao.ReminderDAO;
+import com.lineate.mdzinarishvili.reminderapp.exceptions.InvalidInputException;
+import com.lineate.mdzinarishvili.reminderapp.exceptions.NotFoundException;
+import com.lineate.mdzinarishvili.reminderapp.models.Reminder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
 public class ReminderService {
+    private final ReminderDAO reminderDao;
+
+    public ReminderService(ReminderDAO reminderDao) {
+        this.reminderDao = reminderDao;
+    }
+
+    public List<Reminder> getReminders() {
+        return reminderDao.selectReminders();
+    }
+
+    public Reminder addNewReminder(Reminder reminder) {
+        return reminderDao.insertReminder(reminder);
+    }
+
+    public Reminder updateReminder(Reminder reminder) {
+        if(!reminderDao.isIdValid(reminder.getId())){

```

Comment:
You should just update reminder here inside the try-catch block and if there is any exception then throw IllegalArgumentException with the caught exception message 
Your dao layer should throw database exceptions